File: Two pointer/11. Container With Most Water.cpp
class Solution {
  public:
      int maxArea(vector<int>& height) {
          int left = 0;
          int right = height.size() - 1;
          int ans = INT_MIN;
          while (left < right) {
              int cur = 0;
              if (height[left] < height[right]) {
                  cur = height[left] * (right - left);
                  left++;
              } else {
                  cur = height[right] * (right - left);
                  right--;
              }
              if (cur > ans) {
                  ans = cur;
              }
          }
          return ans;
      }
  };


  class Solution {
    public:
        int maxArea(vector<int>& height) {
            int left = 0;
            int right = height.size() - 1;
            int max_volumn = 0;
            while (left < right) {
                int cur_volumn = (right - left) * min(height[left], height[right]);
                height[left] < height[right] ? left++ : right--;
                max_volumn = max(cur_volumn, max_volumn);
            }
            return max_volumn;
        }
    };

File: Two pointer/11. Container With Most Water/11. Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        start = 0 
        end = len(height) - 1
        max_volume = 0

        while start < end:
            cur_volume = min(height[start], height[end]) * (end - start)
            max_volume = max(max_volume, cur_volume)
            if height[start] < height[end]:
                start += 1
            else: 
                end -= 1

        return max_volume